docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -82,30 +82,122 @@ a :ref:`connector-page-sink-provider`.
 .. _connector-split-manager:
 
 ConnectorSplitManager
-^^^^^^^^^^^^^^^^^^^^^
+---------------------
 
 The split manager partitions the data for a table into the individual
 chunks that Trino will distribute to workers for processing.
 For example, the Hive connector lists the files for each Hive
-partition and creates one or more split per file.
+partition and creates one or more splits per file.
 For data sources that don't have partitioned data, a good strategy
 here is to simply return a single split for the entire table. This
 is the strategy employed by the Example HTTP connector.
 
 .. _connector-record-set-provider:
 
 ConnectorRecordSetProvider
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+--------------------------
 
 Given a split and a list of columns, the record set provider is
 responsible for delivering data to the Trino execution engine.
+
 It creates a ``RecordSet``, which in turn creates a ``RecordCursor``
 that is used by Trino to read the column values for each row.
 
+The provided record set must only include requested columns
+in the correct order. If it returns more rows than necessary,
+the engine will filter them out unless the connector implements
+pushing down predicates, limit, or top-N, and informs the engine
+that this is not necessary.
+
+For simple connectors, where performance is not critical,
+the record set provider can return an instance of ``InMemoryRecordSet``.
+It can be built using lists of values for every row, which can be
+simpler than implementing a ``RecordCursor``.
+
+A ``RecordCursor`` implementation needs to keep track of the current record.
+It return values for columns by a numerical position, in the data type matching
+the column definition in the table. When the engine is done reading the current
+record it will call ``advanceNextPosition`` on the cursor.
+
+Type mapping
+^^^^^^^^^^^^
+
+Both the ``InMemoryRecordSet`` and the ``RecordCursor`` use the following
+Java types:
```

Comment:
The Java type needs to match the carrier type of the corresponding SQL type being returned. 

The list below is incomplete and slightly misleading. It would be better to describe the SQL types, the carrier types and then explain how they relate to the methods in RecordCursor.

Reply:
Thanks! I changed this into a table, listing all built-in SQL types and their carrier types. PTAL. For `HyperLogLog` and others I left it as `Object`, do you know what are their carrier types or where to find them?

Reply:
You need to look at the output of `Type.getJavaType()`

Reply:
Thanks, updated!